WORKSPACE.bazel

```diff
@@ -130,3 +138,64 @@ new_local_repository(
     build_file = "//third_party:system_libraries.BUILD",
     path = "/",
 )
+
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+
+git_repository(
```

Comment:
Can we move python specific dependencies into a `python_repositories.bzl` file? maybe under `tools/bazel_repositories`. (I know the other repository bzl files are at the root, but I think I want to move there somewhere more contained :D )

Reply:
I will do it.

Reply:
Does it make sense to move all other repositories bzl files into the new directory `tools/bazel_repositories` as well?

Reply:
https://github.com/magma/magma/pull/10394

Reply:
I followed the instructions from #10394 and moved the python-specific dependencies into the `bazel` folder. In the process, I cleaned up the code for the python toolchain.